src/inc/jithelpers.h

```diff
@@ -135,7 +135,7 @@
     JITHELPER(CORINFO_HELP_FIELD_ACCESS_EXCEPTION,JIT_ThrowFieldAccessException, CORINFO_HELP_SIG_REG_ONLY)
     JITHELPER(CORINFO_HELP_CLASS_ACCESS_EXCEPTION,JIT_ThrowClassAccessException, CORINFO_HELP_SIG_REG_ONLY)
 
-#ifdef WIN64EXCEPTIONS
+#ifndef _TARGET_X86_
```

Comment:
@janvorli It is a simple workaround. As JIT is not revised yet, JIT attempts to use JIT_EndCatch even when WIN64EXCEPTIONS is enabled while running simple 'Hello, World' example. This causes assert failure.